src/django_svcs/apps.py

```diff
@@ -2,27 +2,34 @@
 from typing import Protocol, cast
 
 import svcs
+from asgiref.local import Local
 from django.apps import AppConfig, apps
-from django.conf import LazySettings, Settings, settings
+from django.conf import settings
 from django.core.exceptions import ImproperlyConfigured
 from django.http import HttpRequest
 
 _KEY_CONTAINER = "svcs_container"
+_NON_REQUEST_CONTEXT = Local()
 
 
 class DjangoRegistryContainer(Protocol):
     registry: svcs.Registry
 
 
 class SvcsConfig(AppConfig):
-    """Django AppConfig for the services module
+    """
+    Django AppConfig for the services module
+
+    This appconfig creates a registry for services and initializes it
+    with services from the project and all installed apps that provide
+    `services.svcs_init` functions.
 
-    This appconfig creates a registry for services and initializes it with services from the project
-    and all installed apps that provide `services.svcs_init` functions.
+    In order to be fully functional, it's necessary to add request
+    wrapping middleware to your project by including the
+    `django_svcs.middleware.request_container` middleware in your
+    `MIDDLEWARE` setting. It must be before any middleware that
+    accesses services.
 
```

Comment:
This chunk looks like you're running with a different formatter config than I am. Not sure if I missed formatting this, or you overformatted, but ...  it looks like you have line length smaller than I roll with :D 

Lemme check that.

Reply:
If you tell me what you like I can set mine to match.

Also I can fix it for good in a followup PR to try to clean up the testing/linting/formatting/packaging stuff, by dropping in config that most IDEs/editors will be able to pick up.

Reply:
Heh. I am not, in fact, a member of the cult of `.editorconfig` -- I dipped my toe in years ago when I was an Eclipse user, but noped out.

I'll settle for a clean configuration for `ruff` in pyproject.toml that has a 100-char line length, and then my editor "does the right thing".

Reply:
Applied 100-character line length manually. You can config it however you want to, but I'd honestly just do it via editorconfig.